csharp/Arsenal/Arsenal/CombinerDemo/WordCountCombinerDemo.cs

```diff
@@ -0,0 +1,176 @@
+        private WordCountReducer reducer;
+        private WordCountEventQueue queue;
+
+        List<int> bucketProcessingCount;
+
+        public WordCountCombinerDemo()
+        {
+            // config inited outside
+        }
+
+        public void Run(string[] args)
+        {
+            Dictionary<string, string> dict = new Dictionary<string, string>();
+
+            for (int i = 0; i < args.Length; i += 2)
+            {
+                dict[args[i]] = args[i + 1];
+            }
+
+            var properties = GetType().GetProperties();
+            foreach (var property in properties)
+            {
+                string name = property.Name;
+                string value = dict["--" + name];
+                object parsedValue = null;
+                if (property.PropertyType == typeof(string))
+                {
+                    parsedValue = value;
+                }
+                else if (property.PropertyType == typeof(int))
+                {
+                    parsedValue = int.Parse(value);
+                }
+                else if (property.PropertyType == typeof(TimeSpan))
+                {
+                    parsedValue = TimeSpan.Parse(value);
+                }
+
+                property.SetValue(this, parsedValue);
+            }
+
+            Task task = Start();
+            while (!task.IsCompleted)
+            {
+                Thread.Sleep(1000 * 60);
+            }
+        }
+
+        public async Task Start()
+        {
+            reducer = new WordCountReducer(AggregateLevel, WordCountFolder);
+            queue = new WordCountEventQueue(NumBuckets);
+            bucketProcessingCount = Enumerable.Repeat(0, NumBuckets).ToList();
+            reducer.Queue = queue;
+            queue.Reducer = reducer;
+            Running = true;
+
+            reducerTasks = Enumerable.Range(0, NumBuckets).Select(ReduceUntilFinished).ToList();
+            Task generateEventsTask = GenerateEventsTask();
+            
+            await generateEventsTask;
+            Console.WriteLine("generating process finished!");
+
+            Task resetRunning = ResetRunningWhenQueuesAreEmpty();
+            await Task.WhenAll(reducerTasks);
+
+            await resetRunning;
+            Console.WriteLine("all reducing tasks finished!");
+            Console.WriteLine(string.Join(", ", bucketProcessingCount));
+
+        }
+
+        public async Task GenerateEventsTask()
+        {
+            Console.WriteLine("Word generation started!");
+            Random rand = new Random();
+            for (int i = 0; i < OccuranceOfSkewWords; i++)
+            {
+                for (int j = 0; j < TotalDifferentWords; j++)
+                {
+                    if (j >= 5)
+                    {
+                        if (rand.NextDouble() > 0.1)
+                            continue;
+                    }
+
+                    string word = new string((char)((int)'A' + j), 10);
```

Comment:
so we support at most 26 different strings?

Reply:
seems so. Handle it later...